Bottom: a4cd837fd30351e35d69c7eb8c9c7ee80672caae
Top:    e77dbd42bc879dfa9461e68ac97e9adffec0ced8
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-12-19 00:47:35 +0000

heap.c: Don't set the v variable twice to the same value in __upheap.


---

diff --git a/src/heap.c b/src/heap.c
index 9e0f35e..57a8208 100644
--- a/src/heap.c
+++ b/src/heap.c
@@ -159,7 +159,6 @@ __upheap (heap_t * heap, int k)
   assert (heap && heap->size < heap->max);
   assert (k <= heap->size);
 
-  v = heap->element[k];
   heap->element[0].cost = MIN_COST;
   for (v = heap->element[k]; heap->element[k / 2].cost > v.cost; k = k / 2)
     heap->element[k] = heap->element[k / 2];
